### DISCUSSION TITLE
20241205 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#125] [Meetings]
- Time:  9:30 in Canary Islands (WET (UTC+0)).
- Time: 10:30 in Vaduz (CET (UTC+1)).

Location:
<https://meeting.blockfinance-eco.li/NautilusCyberneeringWeeklyDevMeeting>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- Cameron Garnham(@da2ce7)  - x 
- Jose Celano(@josecelano) - x
- Constantin Bosse(@cgbosse) - x
- Mario Melchor(@mario-nt) - x
- Graeme Byrne(@grmbyrn) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Summary

My main focus was:

- Reply to @da2ce7's [proposal about refactoring Index configuration](https://github.com/torrust/torrust-index/issues/778) to allow dynamic (DB settings) settings.
- Refine [my proposal](https://github.com/torrust/torrust-index/issues/775#issuecomment-2507555068) for creating the `root` user in the Index.
- Refactor the [Beencode2Json](https://github.com/torrust/bencode2json) package to extract the tokenizer and include feedback from @da2ce7.

### Torrust Website

PR reviews: 

- https://github.com/torrust/torrust-website/pull/129

### Torrust Tracker

Issues:

- https://github.com/torrust/torrust-tracker/issues/1097 (it has been opened again because there are more slow tests on @mario-nt's laptop).

PR reviews: 

- https://github.com/torrust/torrust-tracker/pull/1105. I created and merged a new [PR](https://github.com/torrust/torrust-tracker/pull/1114).
- https://github.com/torrust/torrust-tracker/pull/1114

PRs:

- [Update some (easy-to-update) dependencies](https://github.com/torrust/torrust-tracker/pull/1115)

### Torrust Index

Commented on issues:

- https://github.com/torrust/torrust-index/issues/778#issuecomment-2514178518
- https://github.com/torrust/torrust-index/issues/775#issuecomment-2507555068

PRs:

- [Update some (easy-to-update) dependencies](https://github.com/torrust/torrust-index/pull/779)

### Torrust Index GUI

Commented on issues:

- https://github.com/torrust/torrust-index-gui/issues/664#issuecomment-2507659467

### Bencode2Json

Refator:

- https://github.com/torrust/bencode2json/pull/14

Open new issues:

- https://github.com/torrust/bencode2json/issues/15
- https://github.com/torrust/bencode2json/issues/16
- https://github.com/torrust/bencode2json/issues/17
- https://github.com/torrust/bencode2json/issues/18
- https://github.com/torrust/bencode2json/issues/19
- https://github.com/torrust/bencode2json/issues/20

### Torrust Hash2Torrent

Issues:

- https://github.com/torrust/torrust-hash2torrent/issues

**Note:** time spent on Torrust (23h)

### Events

I went to [GofiGeeks Dev Party 2024](https://www.meetup.com/gofigeeks/events/304581127)

### Discussions

After our discussions in the [20241128 Weekly Development Meeting](https://github.com/orgs/nautilus-cyberneering/discussions/124), I have reviewed these three issues, including the team's feedback and more ideas and details:

- https://github.com/torrust/torrust-index-gui/issues/664
- https://github.com/torrust/torrust-index/issues/775
- https://github.com/torrust/torrust-index-gui/issues/669

I would like to review them again in this meeting to be sure we all agree because they will introduce significant changes, and reverting them in the future could be very costly.

cc @da2ce7 @mario-nt

**Comment by mario-nt** [comment]
### Tracker
- https://github.com/torrust/torrust-tracker/issues/1069

### Review

- https://github.com/torrust/torrust-index/issues/775#issuecomment-2498790001
- https://github.com/torrust/torrust-index/issues/778
- https://github.com/torrust/torrust-tracker/issues/1097#event-15478025002

### Other
- Upgrade PC RAM research
- Fix PC freezing when running Tracker's Doc tests
- 1 to 1 with Jose
